firedrake/adjoint/ensemble_reduced_functional.py

```diff
@@ -88,10 +153,31 @@ def derivative(self, adj_input=1.0, options=None):
         --------
         :meth:`~.ensemble.Ensemble.allreduce`, :meth:`pyadjoint.ReducedFunctional.derivative`.
         """
-        dJdm_local = super(EnsembleReducedFunctional, self).derivative(adj_input=adj_input, options=options)
-        dJdm_local = Enlist(dJdm_local)
-        dJdm_total = []
+
+        if self.gather_functional:
+            dJg_dmg = self.gather_functional.derivative(adj_input=adj_input,
+                                                        options=options)
+            i = self.ensemble.ensemble_comm.rank
+            if isinstance(self.functional, list):
+                # we will pack the derivatives into a flattened list
+                dJdm_local = []
+                k = int(np.sum(self.sizes[:i])-1)
+                for j in range(self.sizes[i]):
+                    k += 1
+                    adj_input = dJg_dmg[k]
+                    der = self.Jhats[j].derivative(adj_input=adj_input,
+                                                   options=options)
```

Comment:
```suggestion
                    der = self.Jhats[j].derivative(adj_input=adj_input, options=options)
```

Reply:
I'm not sure why you want me to do that?